src/coreclr/jit/gentree.cpp

```diff
@@ -29442,6 +29442,57 @@ bool GenTree::IsInvariant() const
     return OperIsConst() || OperIs(GT_LCL_ADDR) || OperIs(GT_FTN_ADDR);
 }
 
+//-------------------------------------------------------------------
+// IsVectorPerElementMask: returns true if this node is a vector constant per-element mask
+// (every element has either all bits set or none of them).
+//
+// Returns:
+//     True if this node is a vector constant per-element mask
+//
+bool GenTree::IsVectorPerElementMask(var_types simdBaseType, unsigned simdSize) const
+{
+#ifdef FEATURE_SIMD
+    if (IsCnsVec())
+    {
+        const GenTreeVecCon* vecCon = AsVecCon();
+        if (vecCon->IsAllBitsSet() || vecCon->IsZero())
```

Comment:
do we need these checks? I doubt they save any measurable TP

Reply:
👍, these functions just loop the memory checking if all elements are a given value

Reply:
@tannergooding suggested that these checks should be in place here: https://discord.com/channels/143867839282020352/312132327348240384/1255568916151930990
However ConditionalSelect with all bits set/all bits zero as a mask is pretty useless, so we can get rid of these

Reply:
The consideration is rather just that main loop below already handles this and these checks aren't "cheap", they are just doing a loop and comparison themselves.

So by removing these explicit checks, we can loop once rather than 3 times and still handle all the scenarios.